lib/pmdtester/builders/summary_report_builder.rb

```diff
@@ -0,0 +1,146 @@
+require_relative './html_report_builder'
+require_relative '../pmd_branch_detail'
+
+module PmdTester
+  # Building summary report to show the details about projects and pmd branchs
+  class SummaryReportBuilder < HtmlReportBuilder
+    REPORT_DIR = 'target/reports/diff'.freeze
+    BASE_CONFIG_PATH = 'target/reports/diff/base_config.xml'.freeze
+    PATCH_CONFIG_PATH = 'target/reports/diff/patch_config.xml'.freeze
+    INDEX_PATH = 'target/reports/diff/index.html'.freeze
+
+    def build(projects, base_name, patch_name)
+      @projects = projects
+      @base_details = get_branch_details(base_name)
+      @patch_details = get_branch_details(patch_name)
+
+      FileUtils.mkdir_p(REPORT_DIR) unless File.directory?(REPORT_DIR)
+      index = File.new(INDEX_PATH, 'w')
+
+      html_report = build_html_report('Summary report')
+      copy_css(REPORT_DIR)
+
+      index.puts html_report
+      index.close
+
+      puts 'Built summary report successfully!'
+    end
+
+    def get_branch_details(branch_name)
+      details = PmdBranchDetail.new(branch_name)
+      details.load
+      details
+    end
+
+    def build_body(doc)
+      build_branch_details_section(doc)
+      build_projects_section(doc)
+    end
+
+    def build_branch_details_section(doc)
+      doc.div(class: 'section', name: 'Branch details') do
+        doc.h2 'Branch details:'
+        build_branch_details_table(doc)
+      end
+    end
+
+    def build_branch_details_table(doc)
+      doc.table(class: 'bodyTable', border: '0') do
+        build_branch_table_head(doc)
+        build_branch_table_body(doc)
+      end
+    end
+
+    def build_branch_table_head(doc)
+      doc.thead do
+        doc.tr do
+          doc.th 'Item'
+          doc.th 'base'
+          doc.th 'patch'
+        end
+      end
+    end
+
+    def build_branch_table_body(doc)
```

Comment:
I just got an idea, what could be useful: github allows to craft urls, that shows you the diff between two commits/branches, e.g. https://github.com/pmd/pmd/compare/master...adangel:issue-1186
So, in general: https://github.com/pmd/pmd/compare/#{base_branch}...#{patch_branch}
On the other hand, when we go on to integrate this in pull request builds, we probably can provide a link to the original pull request.